odoo/addons/base/models/ir_qweb.py

```diff
@@ -35,24 +51,29 @@ class IrQWeb(models.AbstractModel, QWeb):
     _name = 'ir.qweb'
     _description = 'Qweb'
 
+    _empty_line = re.compile(r'\n\s*\n')
+
     @api.model
     def _render(self, id_or_xml_id, values=None, **options):
         """ render(id_or_xml_id, values, **options)
 
         Render the template specified by the given name.
 
-        :param id_or_xml_id: name or etree (see get_template)
+        :param id_or_xml_id: name or etree (see _get_template)
         :param dict values: template values to be used for rendering
         :param options: used to compile the template (the dict available for the rendering is frozen)
             * ``load`` (function) overrides the load method
-            * ``profile`` (float) profile the rendering (use astor lib) (filter
-              profile line with time ms >= profile)
+            * ``profile`` (Boolean) activate the rendering profile displayed in log as debug
         """
-
         context = dict(self.env.context, dev_mode='qweb' in tools.config['dev_mode'])
         context.update(options)
 
-        result = super(IrQWeb, self)._render(id_or_xml_id, values=values, **context)
+        rendering = super(IrQWeb, self)._render(id_or_xml_id, values=values, **context)
+
+        joined = ''.join(rendering)
```

Comment:
qweb.py became a generator of string (to be linked directly to http and returned before rendering is complete as requested by fp)

Reply:
> qweb.py became a generator of string (to be linked directly to http and returned before rendering is complete as requested by fp)

What's that supposed to do that'd be useful? We already delay template rendering to `get_response`. And since the rendering requires an active request with a living cursor and we need to perform post-processing operations on the response, it's not like we will be able to shove the data through to WSGI.

Does this do anything other than make it more difficult to profile template rendering callers by assigning the actual rendering costs onto whoever ends up needing to reify the render, and triggering more hard-to-reproduce bugs?

Keep in mind: if we use generators, this means literally nothing happens until the moment the template output gets iterated. This means e.g. the rendering context can be updated in-place afterwards and the template rendering will use the values which didn't exist when it was rendered.

> It is necessary to remove the part of the strange post treatment of the tables with the station wagon for that.

I've no idea what this is supposed to say. But anyway this makes the override broken, or the original version broken (and this breakage hidden by the override): the parent class returns an iterable of strings, this fixes it to return a single bytes object.